### PR TITLE
Collapsible column groups

### LINKED ISSUE #3343: Collapsible column groups feature
### Description  
Implement collapsible column group feature.

[Example](https://www.infragistics.com/community/blogs/b/igniteui_team/posts/iggrid-collapsible-column-groups-feature), igGrid feature.

### PR DESCRIPTION
**Pull request by ddincheva**
Closes #3343  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [x] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by zdrawku** [comment]
@DaniMarinov and @StefanIvanov, both of you were involved in the Spec creation, so could you take a look at the expand/collapse indicator and verify that everything is okay there.

**Comment by StefanIvanov** [comment]
The spec has been updated with the desired style and position of the expand indicators, we can discuss the updates to the spec if necessary.

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 1]
```diff
@@ -1335,6 +1335,33 @@
         }
     }
 
+
+    %igx-grid__th-expander {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        margin-right: rem(8px);
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: Keep RTL in mind.

File: projects/igniteui-angular/src/lib/grids/columns/column-group.component.ts [hunk 2]
```diff
@@ -26,6 +26,54 @@ export class IgxColumnGroupComponent extends IgxColumnComponent implements After
...
+     *  <igx-column-group [collapsible] = "true"></igx-column-group>
+     * ```
+     * @memberof IgxColumnGroupComponent
+     */
+    @Input()
+    public set collapsible(value: boolean) {
+        this._collapsible = value;
+        this.collapsibleChange.emit(this._collapsible);
+        if (this.children && !this.hidden) {
+            if (value) {
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/grids/columns/column-group.component.ts]:
```suggestion
            if (this._collapsible) {
```
A listener may change the value of `_collapsible` on emit.

File: projects/igniteui-angular/src/lib/grids/columns/column-group.component.ts [hunk 3]
```diff
@@ -143,7 +203,14 @@ export class IgxColumnGroupComponent extends IgxColumnComponent implements After
     set hidden(value: boolean) {
         this._hidden = value;
         this.hiddenChange.emit(this._hidden);
-        this.children.forEach(child => child.hidden = value);
+        if (value || !this.collapsible) {
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/grids/columns/column-group.component.ts]:
```suggestion
        if (this._hidden || !this.collapsible) {
```

File: projects/igniteui-angular/src/lib/grids/columns/column.component.ts [hunk 4]
```diff
@@ -1026,6 +1043,42 @@ export class IgxColumnComponent implements AfterContentInit {
...
+     * @memberof IgxColumnComponent
+     */
+    @Input()
+    set visibleWhenCollapsed(value: boolean) {
+        this._visibleWhenCollapsed = value;
+        this.visibleWhenCollapsedChange.emit(this._visibleWhenCollapsed);
+        if (this.parent) { this.parent.setExpandCollapseState(); }
+        if (this.grid) {
+            this.grid.notifyChanges(true);
+        }
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/grids/columns/column.component.ts]: We have a decorator for this: `@notifyChanges`

File: projects/igniteui-angular/src/lib/grids/columns/column.component.ts [hunk 5]
```diff
@@ -1026,6 +1043,42 @@ export class IgxColumnComponent implements AfterContentInit {
...
+     * Indicates whether the column will be visible when its parent is collapsed.
+     * ```html
+     * <igx-column-group>
+     *   <igx-column [visibleWhenCollapsed]="true"></igx-column>
+     * </igx-column-group>
+     * ```
+     * @memberof IgxColumnComponent
+     */
+    @Input()
+    set visibleWhenCollapsed(value: boolean) {
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/grids/columns/column.component.ts]:
```suggestion
   @Input()
   @notifyChanges(true)
    set visibleWhenCollapsed(value: boolean) {
```

**Pull request merged**